File: kyonggiyo-core/src/main/java/kyonggiyo/domain/candidate/Status.java
package kyonggiyo.domain.candidate;

import kyonggiyo.common.exception.GlobalErrorCode;
import kyonggiyo.common.exception.InvalidValueException;

import java.util.Arrays;

public enum Status {
    ACCEPTED,
    WAITING,
    ;

    public static Status from(String status) {
        return Arrays.stream(Status.values())
                .filter(v -> v.name().equals(status.toUpperCase()))
                .findAny()
                .orElseThrow(() -> new InvalidValueException(GlobalErrorCode.INVALID_REQUEST_EXCEPTION));
    }

}

File: kyonggiyo-core/src/main/java/kyonggiyo/application/candidate/domain/vo/Status.kt
package kyonggiyo.application.candidate.domain.vo

import kyonggiyo.common.exception.GlobalErrorCode
import kyonggiyo.common.exception.InvalidValueException

enum class Status {
    ACCEPTED,
    WAITING;

    companion object {
        fun from(status: String): Status {
            return values().find { it.name == status.uppercase() }
                ?: throw InvalidValueException(GlobalErrorCode.INVALID_REQUEST_EXCEPTION)
        }
    }

}
